### PR TITLE
Update regression readme

### PR DESCRIPTION
**Pull request by DyrellC**
# Description

Update the regression test documentation with instructions on how to run locally and through TIAB. 

## Type of change
- Documentation Fix

# Checklist:
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: python-regression/README.md [hunk 1]
```diff
@@ -1,33 +1,140 @@
...
+contains several signed milestone transactions, and several unsigned transactions. One node is set to validate the 
+testnet coordinator signature, while the other is not. The one that requires validation should solidify to one point, 
+while the other should solidify further. 
+
+Machine 6 - Local Snapshotting Tests: This machine uses 4 nodes. The first node contains the snapshot `meta` and `state`
+files, the `spent-addresses-db` and `testnetdb` of a synced node. The second only contains the database, and the third 
+only contains the snapshot files. All three of these nodes are tested to ensure that they solidify to the same point, 
+and that the proper information is contained in the snapshot files and databases. The fourth node has a larger database 
+that contains more milestones than the `local-snapshots-pruning-depth`. This node is checked to make sure that after 
+starting, transactions that should be pruned form the database have been pruned correctly.
```
Comment (kwek20) [inline python-regression/README.md]: form -> from

File: python-regression/README.md [hunk 2]
```diff
@@ -1,33 +1,140 @@
...
+starting, transactions that should be pruned form the database have been pruned correctly.   
+
+### Running Tests on TIAB (Requires present kubernetes configuration)
+These tests have been written to interact with a set of nodes with provided configurations, running on a `kubernetes` 
+cluster. The `ciglue.sh` script will run all the available machines in the `./tests/features/` directory. The nodes will
+be configured based off of the `./tests/features/machine[x]/config.yml` file, and will update the 
+`./tests/features/machine[x]/output.yml` file with the newly created cluster host and port information. This script then
+proceeds to run each of the feature files in the respective machine directories, and these tests are run on the nodes 
+specified in the `output.yml` file. Once the tests have completed, the cluster is torn down. The default logging 
+of these tests is verbose, for debugging purposes. To use this automated script, simply use the following command:
```
Comment (kwek20) [inline python-regression/README.md]: Can we change the default logging behaviour? (Options for -v)

File: python-regression/README.md [hunk 3]
```diff
@@ -1,33 +1,140 @@
...
+``` 
+
+Once the `output.yml` configuration is set, the tests can be run.
+
+From the `python-regression` directory, a test can be run using the following command structure: 
 ```
-Extra Features
-If you would like a more verbose readout of the tests being run, you can add the -v flag:
+aloe [Feature file name]  -w [Location of feature file]  -v [Optional: verbose test name logging] --nologcapture 
+[Optional: release logging for individual tests for increased verbosity]
```
Comment (GalRogozinski) [inline python-regression/README.md]: The `[....]` looks like something we need to fill in when typing the command in the cli, or Optional according to linux `man`.
Maybe change it to something like:

aloe *Feature File Name*  -w *Location of feature file*  [-v] [--nologcapture]

-v - verbose test name logging
--nologcapture - release logging for individual tests for increased verbosity


In linux `man` there would be underlining and no italics. I just wasn't able to underline in the comments.

File: python-regression/README.md [hunk 4]
```diff
@@ -1,33 +1,140 @@
...
+aloe 1_api_tests.feature -w ./tests/features/machine1/ -v --nologcapture 
 ```
-Certain features will have tags on them that can be used for inclusion or exclusion in testing. For example:
+
+
+
+
+
+### Extra Features
+If you would like to only run part of a machine test, flags can be input into the feature file above the `Scenario: `
```
Comment (GalRogozinski) [inline python-regression/README.md]:
```suggestion
If you would like to only run part of a feature test, flags can be input into the feature file above the `Scenario: ` 
```

File: python-regression/README.md [hunk 5]
```diff
@@ -1,33 +1,140 @@
...
+specified in the `output.yml` file. Once the tests have completed, the cluster is torn down. The default logging 
+of these tests is verbose, for debugging purposes. To use this automated script, simply use the following command: 
+```
+bash ciglue.sh [insert iri docker image here]
+```
+
+To run on the latest IOTA build, run:
+```
+bash ciglue.sh iotacafe/iri-dev:latest
+```
```
Comment (GalRogozinski) [inline python-regression/README.md]: can this work locally just like this without running minikube?
for me it doesn't work...

Anyhows, I don't think it makes sense to invest time in running minikube instructions unless we figure out how to run a subset of tests on that and see how we can debug them.

Make sure you leave a note that the tests will fail if you run them locally like that (unless I am mistaken)

**Comment by GalRogozinski** [comment]
@DyrellC have you seen my review?

**Comment by luca-moser** [comment]
Can you please also mention the exact versions of python, pip etc. you are using? I am having difficulties to run any script using `Python 2.7.16` or `python 3.6.7`

If I try to run the machine6 script I get:
```
aloe 6_local_snapshots_tests.feature -w ./tests/features/machine6/ -v --nologcapture 
Traceback (most recent call last):
  File "/home/luca/.local/bin/aloe", line 10, in <module>
    sys.exit(main())
  File "/home/luca/.local/lib/python3.7/site-packages/aloe/__init__.py", line 34, in main
    Runner(argv)
  File "/home/luca/.local/lib/python3.7/site-packages/aloe/runner.py", line 46, in __init__
    super().__init__(*args, **kwargs)
  File "/home/luca/.local/lib/python3.7/site-packages/nose/core.py", line 121, in __init__
    **extra_args)
  File "/usr/lib/python3.7/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/home/luca/.local/lib/python3.7/site-packages/nose/core.py", line 145, in parseArgs
    self.config.configure(argv, doc=self.usage())
  File "/home/luca/.local/lib/python3.7/site-packages/nose/config.py", line 347, in configure
    self.plugins.begin()
  File "/home/luca/.local/lib/python3.7/site-packages/nose/plugins/manager.py", line 99, in __call__
    return self.call(*arg, **kw)
  File "/home/luca/.local/lib/python3.7/site-packages/nose/plugins/manager.py", line 167, in simple
    result = meth(*arg, **kw)
  File "/home/luca/.local/lib/python3.7/site-packages/aloe/plugin.py", line 64, in begin
    FeatureLoader.find_and_load_step_definitions(feature_dir)
  File "/home/luca/.local/lib/python3.7/site-packages/aloe/fs.py", line 59, in find_and_load_step_definitions
    cls.importer.importFromPath(filename, module_name)
  File "/home/luca/.local/lib/python3.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/luca/.local/lib/python3.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.7/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/luca/projects/iri/python-regression/tests/features/steps/response_handling_steps.py", line 2, in <module>
    from util.test_logic import api_test_logic as api_utils
  File "/home/luca/projects/iri/python-regression/util/test_logic/api_test_logic.py", line 2, in <module>
    from iota import Iota, Address, Tag, TryteString
  File "/home/luca/.local/lib/python3.7/site-packages/iota/__init__.py", line 38, in <module>
    from .api import *
  File "/home/luca/.local/lib/python3.7/site-packages/iota/api.py", line 12, in <module>
    from iota.commands import BaseCommand, CustomCommand, core, \
  File "/home/luca/.local/lib/python3.7/site-packages/iota/commands/__init__.py", line 13, in <module>
    import filters as f
  File "/home/luca/.local/lib/python3.7/site-packages/filters/__init__.py", line 27, in <module>
    from filters.extensions import FilterExtensionRegistry
  File "/home/luca/.local/lib/python3.7/site-packages/filters/extensions.py", line 11, in <module>
    from class_registry import EntryPointClassRegistry
  File "/home/luca/.local/lib/python3.7/site-packages/class_registry/__init__.py", line 5, in <module>
    from .registry import *
  File "/home/luca/.local/lib/python3.7/site-packages/class_registry/registry.py", line 33, in <module>
    class BaseRegistry(with_metaclass(ABCMeta, Mapping)):
  File "/usr/lib/python3/dist-packages/six.py", line 827, in __new__
    return meta(name, bases, d)
  File "/usr/lib/python3.7/abc.py", line 126, in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
TypeError: type() doesn't support MRO entry resolution; use types.new_class()
```

File: python-regression/README.md [hunk 6]
```diff
@@ -1,33 +1,140 @@
...
+
+### Running Tests Locally
+
+To run the tests using a local/custom node, the first step will be to define the nodes that you would like to run the 
+tests on. Each `aloe` test will run off of a given feature file, stored in the `./tests/features/machine[x]` directory.
+The tests rely on an `output.yml` configuration file, which contains the host and port information of the nodes that 
+the test will be run on. This file is generated automatically by the `TIAB` approach, but needs to be generated/filled 
+out manually for local tests. The `output.yml` files are found in the machine directory for the test, and should have a
```
Comment (luca-moser) [inline python-regression/README.md]: Do we need the content of the `config.yml` files inside the `output.yml`?

**Comment by luca-moser** [comment]
I really have difficulties trying to get machine6 tests working locally. I have a setup with 4 nodes and this output.yml file: https://ybin.me/p/69ef4d0d05ea4396#vvaAD5kNfygtQF0Kox8Bz+Xe4fWhdGXVGOOfIVi0jeU=

Using a docker container with Python 3.6.5 I've installed the dependencies via pip as describes in the README and then proceed to trying to execute machine6 tests, however, they fail:
https://ybin.me/p/ffe090f157bd59fb#MVe+p2hiua1T8sLG4PnJ7sMfX3bDqT96fT5clb86khQ=

Any idea what I'm doing wrong? The nodes are running when I try to execute the tests.

File: python-regression/README.md [hunk 7]
```diff
@@ -1,33 +1,162 @@
...
-A testing suite for performance and dynamic testing of IRI
+## Python Regression Testing
+A testing suite for performance and dynamic testing of IRI. These tests currently only work in the `python2.7` 
+environment.
 
-Aloe testing
+### Aloe testing
 This suite uses aloe for performing tests. Once this repository is cloned, make sure you are in the PythonRegression 
-directory, and install the necessary packages using:
+directory. Here you can create a VM to run the tests on and install the necessary packages using:
```
Comment (GalRogozinski) [inline python-regression/README.md]:
```suggestion
directory. Here you can create a virtual environment to run the tests on and install the necessary packages using:
```
`venv` is not a VM :-)

File: python-regression/README.md [hunk 8]
```diff
@@ -1,33 +1,162 @@
...
+
+Machine 6 - Local Snapshotting Tests: This machine uses 4 nodes. The first node contains the snapshot `meta` and `state`
+files, the `spent-addresses-db` and `testnetdb` of a synced node. The second only contains the database, and the third 
+only contains the snapshot files. All three of these nodes are tested to ensure that they solidify to the same point, 
+and that the proper information is contained in the snapshot files and databases. The fourth node has a larger database 
+that contains more milestones than the `local-snapshots-pruning-depth`. This node is checked to make sure that after 
+starting, transactions that should be pruned from the database have been pruned correctly. This machine also includes 
+tests for spent addresses including a test for exporting and merging spent addresses using IXI 
+modules.   
+
```
Comment (GalRogozinski) [inline python-regression/README.md]: Good :-)
We just need to remember to update this per in each test PR

**Pull request merged**